### ISSUE TITLE
Bad return status for module build on kernel: 4.15.0-91-generic (x86_64)

### ISSUE THREAD

**Issue by slrslr** [angristan/wireguard-install#140]
Hello, i have just downloaded and tried to run the Wireguard install script on Ubuntu 18.04.4 LTS and it returned:
Bad return status for module build on kernel: 4.15.0-91-generic (x86_64)
and wg fails to run.

```Hit:1 http://security.ubuntu.com/ubuntu bionic-security InRelease
Hit:2 http://archive.ubuntu.com/ubuntu bionic InRelease
Hit:3 http://archive.ubuntu.com/ubuntu bionic-updates InRelease
Hit:4 http://archive.ubuntu.com/ubuntu bionic-backports InRelease
Reading package lists... Done
Reading package lists... Done
Building dependency tree
Reading state information... Done
iptables is already the newest version (1.6.1-2ubuntu2).
resolvconf is already the newest version (1.79ubuntu10.18.04.3).
The following additional packages will be installed:
  dkms libqrencode3 wireguard-dkms wireguard-tools
Suggested packages:
  menu
The following NEW packages will be installed:
  dkms libqrencode3 qrencode wireguard wireguard-dkms wireguard-tools
0 upgraded, 6 newly installed, 0 to remove and 146 not upgraded.
Need to get 0 B/456 kB of archives.
After this operation, 2461 kB of additional disk space will be used.
Selecting previously unselected package dkms.
(Reading database ... 67358 files and directories currently installed.)
Preparing to unpack .../0-dkms_2.3-3ubuntu9.7_all.deb ...
Unpacking dkms (2.3-3ubuntu9.7) ...
Selecting previously unselected package libqrencode3:amd64.
Preparing to unpack .../1-libqrencode3_3.4.4-1build1_amd64.deb ...
Unpacking libqrencode3:amd64 (3.4.4-1build1) ...
Selecting previously unselected package qrencode.
Preparing to unpack .../2-qrencode_3.4.4-1build1_amd64.deb ...
Unpacking qrencode (3.4.4-1build1) ...
Selecting previously unselected package wireguard-dkms.
Preparing to unpack .../3-wireguard-dkms_1.0.20200611-1ubuntu1~18.04.1_all.deb ...
Unpacking wireguard-dkms (1.0.20200611-1ubuntu1~18.04.1) ...
Selecting previously unselected package wireguard-tools.
Preparing to unpack .../4-wireguard-tools_1.0.20200513-1~18.04.2_amd64.deb ...
Unpacking wireguard-tools (1.0.20200513-1~18.04.2) ...
Selecting previously unselected package wireguard.
Preparing to unpack .../5-wireguard_1.0.20200513-1~18.04.2_all.deb ...
Unpacking wireguard (1.0.20200513-1~18.04.2) ...
Setting up wireguard-tools (1.0.20200513-1~18.04.2) ...
wg-quick.target is a disabled or a static unit, not starting it.
Setting up libqrencode3:amd64 (3.4.4-1build1) ...
Setting up dkms (2.3-3ubuntu9.7) ...
Setting up qrencode (3.4.4-1build1) ...
Setting up wireguard-dkms (1.0.20200611-1ubuntu1~18.04.1) ...
Loading new wireguard-1.0.20200611 DKMS files...
Building for 4.15.0-91-generic
Building initial module for 4.15.0-91-generic
**Error! Bad return status for module build on kernel: 4.15.0-91-generic (x86_64)**
Consult /var/lib/dkms/wireguard/1.0.20200611/build/make.log for more information.
Setting up wireguard (1.0.20200513-1~18.04.2) ...
Processing triggers for libc-bin (2.27-3ubuntu1) ...
Processing triggers for man-db (2.8.3-2ubuntu0.1) ...
* Applying /etc/sysctl.d/10-console-messages.conf ...
kernel.printk = 4 4 1 7
* Applying /etc/sysctl.d/10-ipv6-privacy.conf ...
net.ipv6.conf.all.use_tempaddr = 2
net.ipv6.conf.default.use_tempaddr = 2
* Applying /etc/sysctl.d/10-kernel-hardening.conf ...
kernel.kptr_restrict = 1
* Applying /etc/sysctl.d/10-link-restrictions.conf ...
fs.protected_hardlinks = 1
fs.protected_symlinks = 1
* Applying /etc/sysctl.d/10-lxd-inotify.conf ...
fs.inotify.max_user_instances = 1024
* Applying /etc/sysctl.d/10-magic-sysrq.conf ...
kernel.sysrq = 176
* Applying /etc/sysctl.d/10-network-security.conf ...
net.ipv4.conf.default.rp_filter = 1
net.ipv4.conf.all.rp_filter = 1
net.ipv4.tcp_syncookies = 1
* Applying /etc/sysctl.d/10-ptrace.conf ...
kernel.yama.ptrace_scope = 1
* Applying /etc/sysctl.d/10-zeropage.conf ...
vm.mmap_min_addr = 65536
* Applying /usr/lib/sysctl.d/50-default.conf ...
net.ipv4.conf.all.promote_secondaries = 1
net.core.default_qdisc = fq_codel
* Applying /etc/sysctl.d/99-cloudimg-ipv6.conf ...
net.ipv6.conf.all.use_tempaddr = 0
net.ipv6.conf.default.use_tempaddr = 0
* Applying /etc/sysctl.d/99-sysctl.conf ...
* Applying /etc/sysctl.d/wg.conf ...
net.ipv4.ip_forward = 1
net.ipv6.conf.all.forwarding = 1
* Applying /etc/sysctl.conf ...
Job for wg-quick@wg0.service failed because the control process exited with error code.
See "systemctl status wg-quick@wg0.service" and "journalctl -xe" for details.
Created symlink /etc/systemd/system/multi-user.target.wants/wg-quick@wg0.service → /lib/systemd/system/wg-quick@.service.

WARNING: WireGuard does not seem to be running.
You can check if WireGuard is running with: systemctl status wg-quick@wg0
If you get something like "Cannot find device wg0", please reboot!
```

wg server status:
```Sep 22 19:41:54 ubuntu systemd[1]: Starting WireGuard via wg-quick(8) for wg0...
Sep 22 19:41:54 ubuntu wg-quick[1447]: [#] ip link add wg0 type wireguard
Sep 22 19:41:54 ubuntu wg-quick[1447]: RTNETLINK answers: Operation not supported
Sep 22 19:41:54 ubuntu wg-quick[1447]: Unable to access interface: Protocol not supported
Sep 22 19:41:54 ubuntu wg-quick[1447]: [#] ip link delete dev wg0
Sep 22 19:41:54 ubuntu wg-quick[1447]: Cannot find device "wg0"
Sep 22 19:41:54 ubuntu systemd[1]: wg-quick@wg0.service: Main process exited, code=exited, status=1/FAILURE
Sep 22 19:41:54 ubuntu systemd[1]: wg-quick@wg0.service: Failed with result 'exit-code'.
Sep 22 19:41:54 ubuntu systemd[1]: Failed to start WireGuard via wg-quick(8) for wg0.```

cat /var/lib/dkms/wireguard/1.0.20200611/build/make.log
```DKMS make.log for wireguard-1.0.20200611 for kernel 4.15.0-91-generic (x86_64)
Tue Sep 22 19:40:36 CEST 2020
make: Entering directory '/usr/src/linux-headers-4.15.0-91-generic'
  CC [M]  /var/lib/dkms/wireguard/1.0.20200611/build/main.o
  CC [M]  /var/lib/dkms/wireguard/1.0.20200611/build/noise.o
  CC [M]  /var/lib/dkms/wireguard/1.0.20200611/build/device.o
  CC [M]  /var/lib/dkms/wireguard/1.0.20200611/build/peer.o
  CC [M]  /var/lib/dkms/wireguard/1.0.20200611/build/timers.o
  CC [M]  /var/lib/dkms/wireguard/1.0.20200611/build/queueing.o
  CC [M]  /var/lib/dkms/wireguard/1.0.20200611/build/send.o
  CC [M]  /var/lib/dkms/wireguard/1.0.20200611/build/receive.o
  CC [M]  /var/lib/dkms/wireguard/1.0.20200611/build/socket.o
/var/lib/dkms/wireguard/1.0.20200611/build/socket.c: In function ‘send6’:
/var/lib/dkms/wireguard/1.0.20200611/build/socket.c:139:20: error: ‘const struct ipv6_stub’ has no member named ‘ipv6_dst_lookup_flow’; did you mean ‘ipv6_dst_lookup’?
   dst = ipv6_stub->ipv6_dst_lookup_flow(sock_net(sock), sock, &fl,
                    ^~~~~~~~~~~~~~~~~~~~
                    ipv6_dst_lookup
scripts/Makefile.build:330: recipe for target '/var/lib/dkms/wireguard/1.0.20200611/build/socket.o' failed
make[1]: *** [/var/lib/dkms/wireguard/1.0.20200611/build/socket.o] Error 1
Makefile:1577: recipe for target '_module_/var/lib/dkms/wireguard/1.0.20200611/build' failed
make: *** [_module_/var/lib/dkms/wireguard/1.0.20200611/build] Error 2
make: Leaving directory '/usr/src/linux-headers-4.15.0-91-generic```

**Comment by slrslr** [comment]
After apt update;apt upgrade i seen it updated kernel
old 4.15.0-91
new 4.15.0-118
upgrade shown some relevant lines:
```
make -j1 KERNELRELEASE=4.15.0-118-generic -C /lib/modules/4.15.0-118-generic/build M=/var/lib/dkms/**wireguard**/1.0.20200611/build......
...
writing new private key to '/var/lib/shim-signed/mok/MOK.priv'
-----
 - /var/lib/dkms/wireguard/1.0.20200611/4.15.0-118-generic/x86_64/module/wireguard.ko
EFI variables are not supported on this system
/sys/firmware/efi/efivars not found, aborting.
cleaning build area...

DKMS: build completed.

wireguard.ko:
Running module version sanity check.

Good news! Module version 1.0.20200611 for wireguard.ko
exactly matches what is already found in kernel 4.15.0-118-generic.
DKMS will not replace this module.
You may override by specifying --force.

depmod...

DKMS: install completed.

```
after "reboot", i was in new kernel (uname -r)

and Wireguard was apparently running:
```
# systemctl status wg-quick@wg0
● wg-quick@wg0.service - WireGuard via wg-quick(8) for wg0
   Loaded: loaded (/lib/systemd/system/wg-quick@.service; indirect; vendor preset: enabled)
   **Active: active (exited)** since Tue 2020-09-22 20:01:08 CEST; 42s ago
```

the client .conf files not worked to connect server and when re-run wireguard-install script i have found that there were no active clients (when trying to remove existing) - maybe it not added it due to the installation error mentioned, so i added new one and **it started working..**

**Comment by randomshell** [comment]
Fixed in #125
